raster/src/main/scala/geotrellis/raster/rasterize/CellValue.scala

```diff
@@ -1,5 +1,7 @@
 package geotrellis.raster.rasterize
 
+import geotrellis.raster.CellType
 
-/** Cell value with its zindex used by the rasterizer. */
-case class CellValue(value: Double, zindex: Short)
+
+/** Cell value with its zindex and celltype to be used by the rasterizer. */
+case class CellValue(value: Double, zindex: Double, celltype: CellType)
```

Comment:
I'm realizing now that the issue was misstated, exposing `cellType` for zindex per feature actually leads ambiguous behavior: the order in which tiles are merged is undefined and if two priority tiles do not share the same zindex type it is unclear which will be updated and propagated.

So this parameter should be at function invocation level as `zindexCellType`. Probably `ByteConstantNoDataCellType` is a nice conservative default value for it.